dialect_cockroach.go

```diff
@@ -68,30 +68,28 @@ func (p *cockroach) Create(s store, model *Model, cols columns.Columns) error {
 	keyType := model.PrimaryKeyType()
 	switch keyType {
 	case "int", "int64":
-		cols.Remove("id")
-		id := struct {
-			ID int `db:"id"`
-		}{}
+		cols.Remove(model.IDField())
 		w := cols.Writeable()
 		var query string
 		if len(w.Cols) > 0 {
-			query = fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s) returning id", p.Quote(model.TableName()), w.QuotedString(p), w.SymbolizedString())
+			query = fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s) returning %s", p.Quote(model.TableName()), w.QuotedString(p), w.SymbolizedString(), model.IDField())
 		} else {
-			query = fmt.Sprintf("INSERT INTO %s DEFAULT VALUES returning id", p.Quote(model.TableName()))
+			query = fmt.Sprintf("INSERT INTO %s DEFAULT VALUES returning %s", p.Quote(model.TableName()), model.IDField())
 		}
 		log(logging.SQL, query)
 		stmt, err := s.PrepareNamed(query)
 		if err != nil {
 			return err
 		}
-		err = stmt.Get(&id, model.Value)
+		id := map[string]interface{}{}
+		err = stmt.QueryRow(model.Value).MapScan(id)
 		if err != nil {
 			if err := stmt.Close(); err != nil {
 				return errors.WithMessage(err, "failed to close statement")
 			}
 			return err
 		}
-		model.setID(id.ID)
+		model.setID(id[model.IDField()])
```

Comment:
Is it safe to assume that this entry always exists?

Reply:
This is basically the dynamic version of https://github.com/gobuffalo/pop/pull/590/files#diff-bbab701ece62d45ddcad2772dbb0d7feL71-R71

Reply:
I mean we are always `... returning {{ model.IDField() }}` in the sql query so I guess it is safe to assume it exists.